meson.build

```
@@ -0,0 +1,642 @@
+project('c3c', ['c', 'cpp'])
+cc = meson.get_compiler('c')
+
+if get_option('hide_unused_function_warning')
+    add_global_arguments('-Wno-unused-function', language: 'c')
+    add_global_arguments('-Wno-unused-function', language: 'cpp')
```

```suggestion
    add_global_arguments('-Wno-unused-function', language: ['c', 'cpp'])
```

I would also generally advise to use add_project_arguments instead of add_global_arguments.

---

As a matter of curiosity, why use a project option to choose whether to hide the warnings?

---

I was trying to keep it conventional and configurable when it comes to building project. Also noted, on project arguments, thank you for that.